### PR TITLE
Eslint no sequences

### LINKED ISSUE #3494: Add eslint rules to catch issues flagged by SonarQube
Below is a breakdown of eslint rules that we should add to prevent things that SonarQube considers code smells by their priority in SonarQube.

These rules should be applied and fixed one per PR.

### Critical
* [x] [default-case-last](https://eslint.org/docs/rules/default-case-last) &mdash; #3493 
* [x] [no-void](https://eslint.org/docs/rules/no-void) &mdash; #3509 

### Major
* [x] [no-redeclare](https://eslint.org/docs/rules/no-redeclare) (already included by `eslint:recommended`)
* [x] [no-sequences](https://eslint.org/docs/rules/no-sequences) &mdash; #3609
* [x] [no-shadow](https://eslint.org/docs/rules/no-shadow) &mdash; #4058
* [x] [no-nested-ternary](https://eslint.org/docs/rules/no-nested-ternary) &mdash; #3603
* [ ] [max-params](https://eslint.org/docs/rules/max-params) with a max of 7 &mdash; #4035
* [x] [no-throw-literal](https://eslint.org/docs/rules/no-throw-literal) &mdash; #3798

### Minor
* [x] [no-return-await](https://eslint.org/docs/rules/no-return-await) &mdash; #3796
* [x] [no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)
* [x] [no-duplicate-imports](https://eslint.org/docs/rules/no-duplicate-imports) &mdash; #3539

### PR DESCRIPTION
**Pull request by sancodes**
**Changes**

- remove trailing space at `.eslintrc.js` file after `no-redeclare` declaration
- add `no-sequences` eslint
- ran `npm run lint`
- no issues were found, thus pushing it for merge request

**Issues**

Fixes #3494 `no-sequences` case

### CODE REVIEW

**Comment by sancodes** [comment]
@thornbill  I lost this branch for this PR. is there a way I can recover it? So I can get the issues fixed.

**Comment by dmitrylyzo** [comment]
> @thornbill I lost this branch for this PR. is there a way I can recover it? So I can get the issues fixed.

It is still in your fork.
Assuming you cloned your fork as `origin` remote. Checkout `origin/eslint-no-sequences` with `eslint-no-sequences` name.
```sh
git checkout -b eslint-no-sequences origin/eslint-no-sequences
```

**Comment by sancodes** [comment]
@dmitrylyzo thank you.

**Comment by sancodes** [comment]
@thornbill changes have been made, I'm willing to fix if any mistake. Thank you

File: src/components/favoriteitems.js [hunk 1]
```diff
@@ -145,7 +145,7 @@ import '../elements/emby-itemscontainer/emby-itemscontainer';
             let html = '';
 
             if (result.Items.length) {
-                if (html += '<div class="sectionTitleContainer sectionTitleContainer-cards padded-left">', !layoutManager.tv && options.Limit && result.Items.length >= options.Limit) {
+                if ((html += '<div class="sectionTitleContainer sectionTitleContainer-cards padded-left">', !layoutManager.tv && options.Limit && result.Items.length >= options.Limit)) {
```
Comment (dmitrylyzo) [inline src/components/favoriteitems.js:148]:
```suggestion
                html += '<div class="sectionTitleContainer sectionTitleContainer-cards padded-left">';

                if (!layoutManager.tv && options.Limit && result.Items.length >= options.Limit) {
```

File: src/components/libraryoptionseditor/libraryoptionseditor.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ import template from './libraryoptionseditor.template.html';
         let html = '';
         const elem = page.querySelector('.metadataReaders');
 
-        if (plugins.length < 1) return elem.innerHTML = '', elem.classList.add('hide'), !1;
+        if (plugins.length < 1) return (elem.innerHTML = '', elem.classList.add('hide'), !1);
```
Comment (dmitrylyzo) [inline src/components/libraryoptionseditor/libraryoptionseditor.js:67]:
```suggestion
        if (plugins.length < 1) {
            elem.innerHTML = '';
            elem.classList.add('hide')
            return false;
        }

```

File: src/components/libraryoptionseditor/libraryoptionseditor.js [hunk 3]
```diff
@@ -97,7 +97,7 @@ import template from './libraryoptionseditor.template.html';
     function renderMetadataSavers(page, metadataSavers) {
         let html = '';
         const elem = page.querySelector('.metadataSavers');
-        if (!metadataSavers.length) return elem.innerHTML = '', elem.classList.add('hide'), false;
+        if (!metadataSavers.length) return (elem.innerHTML = '', elem.classList.add('hide'), false);
```
Comment (dmitrylyzo) [inline src/components/libraryoptionseditor/libraryoptionseditor.js:100]:
```suggestion
        if (!metadataSavers.length) {
            elem.innerHTML = '';
            elem.classList.add('hide');
            return false;
        }
```

File: src/components/libraryoptionseditor/libraryoptionseditor.js [hunk 4]
```diff
@@ -550,9 +550,9 @@ import template from './libraryoptionseditor.template.html';
     function getOrderedPlugins(plugins, configuredOrder) {
         plugins = plugins.slice(0);
         plugins.sort((a, b) => {
-            a = configuredOrder.indexOf(a.Name);
-            b = configuredOrder.indexOf(b.Name);
-            return a - b;
+          a = configuredOrder.indexOf(a.Name);
+          b = configuredOrder.indexOf(b.Name);
+          return a - b;
```
Comment (dmitrylyzo) [inline src/components/libraryoptionseditor/libraryoptionseditor.js:555]:
```suggestion
            a = configuredOrder.indexOf(a.Name);
            b = configuredOrder.indexOf(b.Name);
            return a - b;
```

**Review by dmitrylyzo — COMMENTED** [review]
I've added examples (not tested - just visually). Others should be changed in the same way.

File: src/components/libraryoptionseditor/libraryoptionseditor.js (hunk 4 above)
Comment (thornbill) [reply src/components/libraryoptionseditor/libraryoptionseditor.js:555]: The indentation checks are still disabled in a bunch of files.

File: src/components/remotecontrol/remotecontrol.js [hunk 5]
```diff
@@ -646,7 +646,7 @@ export default function () {
     }
 
     function bindToPlayer(context, player) {
-        if (releaseCurrentPlayer(), currentPlayer = player, player) {
+        if ((releaseCurrentPlayer(), currentPlayer = player, player)) {
```
Comment (thornbill) [inline src/components/remotecontrol/remotecontrol.js:649]:
```suggestion
        releaseCurrentPlayer();
        currentPlayer = player;

        if (player) {
```

File: src/controllers/itemDetails/index.js [hunk 6]
```diff
@@ -1790,7 +1790,7 @@ function renderAdditionalParts(page, item, user) {
 function renderScenes(page, item) {
     let chapters = item.Chapters || [];
 
-    if (chapters.length && !chapters[0].ImageTag && (chapters = []), chapters.length) {
+    if ((chapters.length && !chapters[0].ImageTag && (chapters = []), chapters.length)) {
```
Comment (thornbill) [inline src/controllers/itemDetails/index.js:1793]:
```suggestion
    if (chapters.length && !chapters[0].ImageTag) {
        chapters =  [];
    }

    if (chapters.length) {
```

File: src/scripts/libraryBrowser.js [hunk 7]
```diff
@@ -86,7 +86,7 @@ export function getQueryPagingHtml (options) {
     const recordsEnd = Math.min(startIndex + limit, totalRecordCount);
     const showControls = limit < totalRecordCount;
 
-    if (html += '<div class="listPaging">', showControls) {
+    if ((html += '<div class="listPaging">', showControls)) {
```
Comment (thornbill) [inline src/scripts/libraryBrowser.js:89]:
```suggestion
    html += '<div class="listPaging">';

    if (showControls) {
```

File: src/controllers/dashboard/metadataImages.js [hunk 8]
```diff
@@ -43,11 +43,11 @@ import Dashboard from '../../utils/dashboard';
 
     function onSubmit() {
         const form = this;
-        return loading.show(), ApiClient.getServerConfiguration().then(function(config) {
+        return (loading.show(), ApiClient.getServerConfiguration().then(function(config) {
             config.PreferredMetadataLanguage = form.querySelector('#selectLanguage').value;
             config.MetadataCountryCode = form.querySelector('#selectCountry').value;
             ApiClient.updateServerConfiguration(config).then(Dashboard.processServerConfigurationUpdateResult);
-        }), !1;
+        }), !1);
```
Comment (thornbill) [inline src/controllers/dashboard/metadataImages.js:50]:
```suggestion
        });
        return false;
```

**Review by thornbill — CHANGES_REQUESTED** [review]

File: src/controllers/dashboard/metadataImages.js [hunk 9]
```diff
@@ -43,11 +43,11 @@ import Dashboard from '../../utils/dashboard';
 
     function onSubmit() {
         const form = this;
-        return loading.show(), ApiClient.getServerConfiguration().then(function(config) {
+        return (loading.show(), ApiClient.getServerConfiguration().then(function(config) {
```
[the remark this answers is not in the document]
Comment (dmitrylyzo) [reply src/controllers/dashboard/metadataImages.js:46]: Bill, return the Promise

File: src/controllers/itemDetails/index.js [hunk 10]
```diff
@@ -1790,7 +1790,11 @@ function renderAdditionalParts(page, item, user) {
 function renderScenes(page, item) {
     let chapters = item.Chapters || [];
 
-    if (chapters.length && !chapters[0].ImageTag && (chapters = []), chapters.length) {
+    if (chapters.length && !chapters[0].ImageTag) {
+        chapters =  [];
```
Comment (dmitrylyzo) [inline src/controllers/itemDetails/index.js:1794]:
```suggestion
        chapters = [];
```

File: .eslintrc.js [hunk 11]
```diff
@@ -61,7 +61,8 @@ module.exports = {
         'no-var': ['error'],
         'space-before-blocks': ['error'],
         'space-infix-ops': 'error',
-        'yoda': 'error'
+        'yoda': 'error',
+        'no-sequences': ['error']
```
Comment (dmitrylyzo) [inline .eslintrc.js:65]: More errors will be found.
```suggestion
        'no-sequences': ['error', { 'allowInParentheses': false }]
```

**Review by thornbill — CHANGES_REQUESTED** [review]
Looks like a few other files had these too... I can't make recommendations on them directly since they aren't part of the changes, but it should just be a matter of using the second portion of the statement. (I have no idea why the first part is even present since it does nothing in practice.)

For example:
```js
SortBy: (section.types, 'SeriesName,SortName'),
```
becomes
```js
SortBy: 'SeriesName,SortName',
```

File: .eslintrc.js (hunk 11 above)
Comment (thornbill) [reply .eslintrc.js]: I had missed that @dmitrylyzo had already made suggestions here for the remaining issues found by eslint.

**Review by thornbill — APPROVED** [review]

**Pull request merged**